ouroboros-consensus-test/bench-mempool/Main.hs

```diff
@@ -37,20 +38,100 @@ main = do
     convertCsvRowsToJsonObjects rawValues "mempool-benchmarks.json"
   where
     runBenchmarks csvFilePath = do
-        opts <- parseOptions benchIngredients benchmarkJustAddingTransactions
+        opts <- parseOptions customBenchIngredients benchmarkJustAddingTransactions
         let opts' = changeOption (Just . fromMaybe (CsvPath csvFilePath)) opts
-        case tryIngredients benchIngredients opts' benchmarkJustAddingTransactions of
+        case tryIngredients customBenchIngredients opts' benchmarkJustAddingTransactions of
           Nothing               -> exitFailure
           Just    runIngredient -> do
             success <- runIngredient
             unless success exitFailure
       where
         benchmarkJustAddingTransactions =
-            bgroup "Just adding" $
-                fmap benchAddNTxs [10_000, 1_000_000]
+            askOption $ \(fromOptionBSS -> bss) ->
+            askOption $ \(getNs -> ns) ->
+            bgroup "Just adding" [
+                bgroup "scenario1" [
```

Comment:
What should we name all these scenarios?

Reply:
We could either leave it as is, but add a space before the number (eg `"scenario 1`), or describe each scenario in terms of what the backing store contains, what the changelog consumes, and what the candidate transactions consume, eg:

```haskell
bgroup "empty        , empty              , n linked"

bgroup "empty        , 2.160 linked       , n linked"

bgroup "empty        , empty              , n independent"

bgroup "n independent, empty              , consume all from bs"

bgroup "n independent, consume all from bs, consume all from cl"
```

In the latter proposal, we need to add a comment that describes what each option in the `bgroup` strings means.

The latter proposal also suggested me another alternative for how to express this DSL, but I think we should not explore further :woozy_face: